### PR TITLE
*:process approximate rows in region

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi
     This PR process the approximate rows in the region from the heartbeat request.
@nolouch @disksing PTAL

### CODE REVIEW

File: server/cluster_info.go [hunk 1]
```diff
@@ -450,6 +450,9 @@ func (c *clusterInfo) handleRegionHeartbeat(region *core.RegionInfo) error {
 		if region.ApproximateSize != origin.ApproximateSize {
 			saveCache = true
 		}
+		if region.ApproximateRows != origin.ApproximateRows {
+			saveCache = true
```
Comment (nolouch) [inline server/cluster_info.go]: Is the `ApproximateRows` too sensitive to change?

File: server/schedule/basic_cluster.go [hunk 2]
```diff
@@ -86,8 +86,10 @@ func (m OpInfluence) GetRegionsInfluence() map[uint64]*Operator {
 // StoreInfluence records influences that pending operators will make.
 type StoreInfluence struct {
 	RegionSize  int64
+	RegionRows  int64
 	RegionCount int64
 	LeaderSize  int64
+	LeaderRows  int64
```
Comment (disksing) [inline server/schedule/basic_cluster.go]: I think we can exclude `RegionRows` and `LeaderRows` in `StoreInfluence`.

File: pkg/testutil/operator_check.go [hunk 3]
```diff
@@ -1,100 +1,100 @@
...
-	c.Assert(op, check.NotNil)
-	c.Assert(op.Len(), check.Equals, 4)
-	c.Assert(op.Step(0), check.FitsTypeOf, schedule.AddLearner{})
-	c.Assert(op.Step(1), check.FitsTypeOf, schedule.PromoteLearner{})
-	c.Assert(op.Step(2), check.FitsTypeOf, schedule.TransferLeader{})
-	c.Assert(op.Step(3).(schedule.RemovePeer).FromStore, check.Equals, sourceID)
-	kind |= (schedule.OpRegion | schedule.OpLeader)
-	c.Assert(op.Kind()&kind, check.Equals, kind)
-}
+// Copyright 2017 PingCAP, Inc.
```
[the remark this answers is not in the document]
Comment (AndreMouche) [reply pkg/testutil/operator_check.go]: Since I do `gofmt` with `go version go1.10.1 darwin/amd64` ?

[the remark this answers is not in the document]
Comment (AndreMouche) [reply pkg/testutil/operator_check.go]: Since I do `gofmt` with `go version go1.10.1 darwin/amd64` ?

File: server/cluster_info.go (hunk 1 above)
Comment (AndreMouche) [reply server/cluster_info.go]: I'm not sure, should we @nolouch @disksing ?

File: pkg/testutil/operator_check.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (disksing) [reply pkg/testutil/operator_check.go]: It looks like that old file is not well formatted. (apparently `make check` skips all files in `testutil`, no idea why...)

File: server/cluster_info.go (hunk 1 above)
Comment (disksing) [reply server/cluster_info.go]: I think it should be ok as long as they are only saved to cache (not to etcd).

**Comment by disksing** [comment]
LGTM.

**Pull request merged**